src/backend/distributed/transaction/backend_data.c

```diff
@@ -1131,3 +1199,57 @@ DecrementClientBackendCounter(void)
 {
 	pg_atomic_sub_fetch_u32(&backendManagementShmemData->activeClientBackendCounter, 1);
 }
+
+
+/*
+ * EndGlobalProcess gets a global pid and and ends the original query with the global pid
+ * that might have started in another node by connecting to that node and running either
+ * pg_cancel_backend or pg_terminate_backend based on the withTerminate argument.
+ */
+static bool
+EndGlobalProcess(uint64 globalPID, bool withTerminate)
+{
+	int nodeId = ExtractNodeIdFromGlobalPID(globalPID);
+	int processId = ExtractProcessIdFromGlobalPID(globalPID);
+
+	List *workerList = ActivePrimaryNodeList(NoLock);
+	WorkerNode *workerNode = NULL;
+
+	char *workerName = NULL;
+	int workerPort = 0;
+
+	foreach_ptr(workerNode, workerList)
+	{
+		if (workerNode->nodeId != nodeId)
+		{
+			continue;
+		}
+
+		workerName = workerNode->workerName;
+		workerPort = workerNode->workerPort;
+		break;
+	}
+
+	StringInfo cancelQuery = makeStringInfo();
+
+	if (withTerminate)
+	{
+		appendStringInfo(cancelQuery, "SELECT pg_terminate_backend(%d)", processId);
+	}
+	else
+	{
+		appendStringInfo(cancelQuery, "SELECT pg_cancel_backend(%d)", processId);
+	}
+
+	StringInfo queryResult = makeStringInfo();
+
+	bool success = ExecuteRemoteQueryOrCommand(workerName, workerPort, cancelQuery->data,
```

Comment:
I think our experience is not the same with Postgres' experience. In Postgres, you may get ERRORs:
```SQL
SELECT pg_terminate_backend(64631);
ERROR:  must be a superuser to terminate superuser process
```

Whereas, we only return false.
```SQL
SELECT pg_terminate_backend(10000064631);
┌──────────────────────┐
│ pg_terminate_backend │
├──────────────────────┤
│ f                    │
└──────────────────────┘
(1 row)

Time: 24.687 ms
```

In general, I think we should have the same experience with Postgres. If the throws an error, we should also throw error, if it returns false/true, we should do the same.

`ExecuteRemoteQueryOrCommand` mostly does what we need. It already knows whether the command throw an error or not. If it throw error, we should probably get the same error message and do something liike:
```
ERROR:  must be a superuser to terminate superuser process
CONTEXT: While executing `SELECT pg_cancel_backend(XXX) on node host:port`
DETAIL: Failed to terminate Citus process with global_pid (%d)
```